### DISCUSSION TITLE
Welcome to flipperzero-bruteforce Discussions!

### DISCUSSION THREAD

**Discussion by tobiabocchi** [tobiabocchi/flipperzero-bruteforce#26] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Onion-btc** [comment]
hello buddy came across this, so would i be able to bruteforce a fob entry system?

**Reply by tobiabocchi** [reply]
Hello! As of right now, this repo is for bruteforcing sub-ghz RF protocols, I think by fob system you mean system that use a tag of some sort (i.e. passive RFID) so the answer is no (at least for now)

**Reply by kanawai** [reply]
The new flipper just arrived, What's the best way to set up my flipper to emulate capture and emulate RKE?

**Reply by kanawai** [reply]
For testing, I live next to parking lot. Much of the frequencies that come in are 309MHz, 315MHz 433MHz, 434MHz, 914MHz, 868MHz. FYI

**Comment by KK4SE** [comment]
Is there an established way to brute force a traffic gate arm in a parking lot?

**Comment by 54DB0153NP41** [comment]
Will this brutforce car doors?


"I accidentally locked my only keys in my car. "

**Comment by Saso995** [comment]
Hi first of all thanks for this great work! Then I would like to ask if you are sure about the timing for PT-2240 (if my math is not wrong it would be higher!) 

By the way I also tested some optimization for PT-2240: with a short pulse `300`, long pulse `3x`, pilot period `31x` it works fine, and it seems to work also with just 2 repetition.

**Comment by XtianLi** [comment]
Hi everyone. First, thaks for your greats contributions. 

I would like to know what do I need to do or modify in case I have the last digit of the code, for the button number, but I want to create a brute force dictionary for the serial number. 

In my case, I have this Code> 
Filetype: Flipper SubGhz Key File
Version: 1
Frequency: 433889000
Preset: FuriHalSubGhzPresetOok650Async
Protocol: Princeton
Bit: 24
Key: 00 00 00 00 00 93 21 E8
TE: 372

Knowing that the last number 8, is the key number, and "93 21 E" is the serial number, how do I set the "key_range=range(0x000004, 0xFFFFF4)" to always finish with the number 4. 

I would like to reduce the dictionary size, and time spent running the bruteforce attack. 

Thanks in advance!

**Reply by XtianLi** [reply]
Well, Sorry, I get it with the lines>

```
# Filtrar solo los números que terminen en "4" en su representación hexadecimal
key_range = [key for key in range(0x000004, 0xFFFFF4) if hex(key).endswith('4')]

protocols = [
    Protocol(
        name="Bike-VZero-433",
        n_bits=24,
        transposition_table={"0": "370 -1090 ", "1": "1090 -370 "},
        pilot_period="345 -11300  ",
        key_range=key_range,
    ),
]

for p in protocols:
    p.generate_sub_files()
    print(f"{p.name} done")
```
    
    
 But now the file generated says not being parsed on the Flipper, and I cant run it. 
 I tried even taking this lines and the file doesnt work. 
 What am I missing?

**Comment by edog245203** [comment]
It won't let me download onto my flipper? Can you give me a step by step. I'm so sorry
